components/suggestion/search_date_suggestion/search_date_suggestion.tsx

```diff
@@ -16,18 +16,19 @@ import 'moment';
 const loadedLocales: Record<string, moment.Locale> = {};
 
 type Props = {
-    locale: string;
+    locale?: string;
+    currentDate?: any;
 }
 
 export default class SearchDateSuggestion extends Suggestion {
     handleDayClick = (day: Date) => {
         const dayString = day.toISOString().split('T')[0];
-        this.props.onClick(dayString, this.props.matchedPretext);
+        this.props.onClick?.(dayString, this.props.matchedPretext);
     }
 
     componentDidMount() {
         //the naming scheme of momentjs packages are all lowercases
-        const locale = this.props.locale.toLowerCase();
+        const locale = this.props.locale && this.props.locale.toLowerCase();
```

Comment:
We can do this one using the coalesce as well.
```suggestion
        const locale = this.props.locale?.toLowerCase();
```